homeassistant/components/qbus/entity.py

```diff
@@ -0,0 +1,83 @@
+"""Base class for Qbus entities."""
+
+from abc import ABC, abstractmethod
+import re
+
+from qbusmqttapi.discovery import QbusMqttOutput
+from qbusmqttapi.factory import QbusMqttMessageFactory, QbusMqttTopicFactory
+from qbusmqttapi.state import QbusMqttState
+
+from homeassistant.components.mqtt import ReceiveMessage, client as mqtt
+from homeassistant.helpers.device_registry import DeviceInfo, format_mac
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, MANUFACTURER
+
+_REFID_REGEX = re.compile(r"^\d+\/(\d+(?:\/\d+)?)$")
+
+
+def format_ref_id(ref_id: str) -> str | None:
+    """Format the Qbus ref_id."""
+    matches: list[str] = re.findall(_REFID_REGEX, ref_id)
+
+    if len(matches) > 0:
+        if ref_id := matches[0]:
+            return ref_id.replace("/", "-")
+
+    return None
+
+
+class QbusEntity(Entity, ABC):
+    """Representation of a Qbus entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        mqtt_output: QbusMqttOutput,
+        entity_id_format: str,
+    ) -> None:
+        """Initialize the Qbus entity."""
+
+        self._topic_factory = QbusMqttTopicFactory()
+        self._message_factory = QbusMqttMessageFactory()
+
+        ref_id = format_ref_id(mqtt_output.ref_id)
+
+        self._attr_unique_id = f"ctd_{mqtt_output.device.serial_number}_{ref_id}"
+        self.entity_id = entity_id_format.format(
+            f"qbus_{mqtt_output.device.serial_number}_{ref_id}"
+        )
+        self._attr_name = mqtt_output.name.title()
+
+        self._attr_device_info = DeviceInfo(
+            name=mqtt_output.name.title(),
```

Comment:
If the name of the device is the same as the entity, set the name of the entity to `_attr_name = None`